servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/ScanWithPublisher.java

```diff
@@ -35,31 +35,7 @@
 
     ScanWithPublisher(Publisher<T> original, Supplier<R> initial, BiFunction<R, ? super T, R> accumulator,
                       Executor executor) {
-        this(original, () -> new ScanWithMapper<T, R>() {
```

Comment:
this change was just to consolidate conversion code outside the constructor (e.g. less noise) and add `requireNonNull` checks. I can break it out into a separate PR but was convenient while I was here and should have no external impact.